ganttproject/src/net/sourceforge/ganttproject/chart/mouse/ChangeTaskProgressRuler.java

```diff
@@ -88,6 +88,34 @@ int getProgress(int pixels) {
     int upperProgress = tailMap.get(upperPx);
 
     float diffProgress = (upperProgress - lowerProgress) * ((float) (pixels - lowerPx) / (float) (upperPx - lowerPx));
-    return (int) (lowerProgress + diffProgress);
+    int overallProgress = (int)(lowerProgress + diffProgress);
+    return createProgress(overallProgress);
+  }
+
+  private Progress createProgress(int progressPercents) {
+    return new Progress() {
+      @Override
+      public int toPercents() {
+        return progressPercents;
+      }
+
+      @Override
+      public String toUnits() {
+        return convertPercentsToUnits(progressPercents);
+      }
+    };
+  }
+
+  String convertPercentsToUnits(int overallProgress) {
+    float taskDuration = myTask.getDuration().getValue();
+    float progressInUnits = overallProgress * taskDuration/100;
+    String wholeUnits = Integer.toString((int)progressInUnits);
+    String fractionIndicator = (overallProgress*myTask.getDuration().getLength()) % 100 == 0 ? "" : "+";
+    return wholeUnits + fractionIndicator;
+  }
+
+  interface Progress {
```

Comment:
I meant interface here as a less powerful version of class with compact syntax not as an API thing, maybe it was misleading. Converted it to abstract class now.